fixit/rules/import_constraints.py

```diff
@@ -328,13 +334,18 @@ def __init__(self, context: CstContext) -> None:
             formatted_config: Dict[
                 Path, Dict[object, object]
             ] = self._parse_and_format_config(import_constraints_config)
-            # Run through logical ancestors of the filepath stopping early if a parent
-            # directory is found in the config. The closest parent's settings will be used.
-            for parent_dir in self._abs_file_path.parents:
-                if parent_dir in formatted_config:
-                    settings_for_dir = formatted_config[parent_dir]
-                    self._config = _ImportConfig.from_config(settings_for_dir)
-                    break
+            # Try matching the filepath and all of its parents paths in any of the
+            # given configurations' glob patterns, stopping early if a match is
+            # found in the config. The closest match in the settings will be used.
+            parts = list(self._abs_file_path.parts)
+            while parts:
+                part_dir = os.path.join(*parts)
+                parts.pop()
+                for pattern, settings_for_dir in formatted_config.items():
+                    if fnmatch(part_dir, pattern):
+                        self._config = _ImportConfig.from_config(settings_for_dir)
+                        parts.clear()
+                        break
```

Comment:
When a match is found, we should also break the `while parts` loop.

Reply:
`parts.clear()` does that.

Reply:
These are the times I wish Python had `break 2`, like PHP does, or `break <label>` like Swift. I can also use the `else: continue; break` idiom if it’s clearer.